File: packages/ruby-sdk/lib/strait/operations/log_drains.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for log drain CRUD.
    class LogDrainsService < BaseService
      # List all log drains.
      # GET /v1/log-drains
      def list(query: nil)
        _request(:get, "/v1/log-drains", query: query)
      end

      # Create a new log drain.
      # POST /v1/log-drains
      def create(body)
        _request(:post, "/v1/log-drains", body: body)
      end

      # Get a log drain by ID.
      # GET /v1/log-drains/{drainID}
      def get(drain_id)
        _request(:get, "/v1/log-drains/{drainID}", path_params: { "drainID" => drain_id })
      end

      # Update a log drain.
      # PATCH /v1/log-drains/{drainID}
      def update(drain_id, body)
        _request(:patch, "/v1/log-drains/{drainID}", path_params: { "drainID" => drain_id }, body: body)
      end

      # Delete a log drain.
      # DELETE /v1/log-drains/{drainID}
      def delete(drain_id)
        _request(:delete, "/v1/log-drains/{drainID}", path_params: { "drainID" => drain_id })
        nil
      end
    end
  end
end

File: packages/python-sdk/strait/operations/log_drains.py
"""Log drain management operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class LogDrainsService(BaseService):
    def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/log-drains", query=query)

    def create(self, body: Any) -> dict[str, Any]:
        return self._request("POST", "/v1/log-drains", body=body)

    def get(self, drain_id: str) -> dict[str, Any]:
        return self._request(
            "GET", "/v1/log-drains/{drainID}", path_params={"drainID": drain_id},
        )

    def update(self, drain_id: str, body: Any) -> dict[str, Any]:
        return self._request(
            "PATCH", "/v1/log-drains/{drainID}", path_params={"drainID": drain_id}, body=body,
        )

    def delete(self, drain_id: str) -> dict[str, Any]:
        return self._request(
            "DELETE", "/v1/log-drains/{drainID}", path_params={"drainID": drain_id},
        )


class AsyncLogDrainsService(AsyncBaseService):
    async def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/log-drains", query=query)

    async def create(self, body: Any) -> dict[str, Any]:
        return await self._request("POST", "/v1/log-drains", body=body)

    async def get(self, drain_id: str) -> dict[str, Any]:
        return await self._request(
            "GET", "/v1/log-drains/{drainID}", path_params={"drainID": drain_id},
        )

    async def update(self, drain_id: str, body: Any) -> dict[str, Any]:
        return await self._request(
            "PATCH", "/v1/log-drains/{drainID}", path_params={"drainID": drain_id}, body=body,
        )

    async def delete(self, drain_id: str) -> dict[str, Any]:
        return await self._request(
            "DELETE", "/v1/log-drains/{drainID}", path_params={"drainID": drain_id},
        )
